pages/setuppwa/setuppwa.tsx

```diff
@@ -0,0 +1,73 @@
+import { useRouter } from "next/router"
+import React from "react"
+
+const SetupPwa = () => {
+  const router = useRouter()
+  const [username, setUsername] = React.useState<string>("")
+
+  let accountUsername: string
```

Comment:
I dont believe this variable is neccary and you should be able to do everything you need to with `username` and `username_from_local`. Additionally you should try and avoid using mutable variables and mutating them in callbacks as it can introduce bugs with references to old variables and not triggering re-renders.